src/hip/_async/sessions.py

```diff
@@ -0,0 +1,111 @@
+import typing
+
+
+HeadersType = typing.Any  # TODO: Replace with the real HeadersType from 'models'
+Request = typing.Any  # TODO: Replace with the real Request type
+AsyncRequestData = typing.Any  # TODO: Replace with the AsyncRequestData type
+Cookies = typing.Any  # TODO: Replace with real Cookies type
+
+QueryType = typing.Union[
+    typing.Sequence[typing.Tuple[str], typing.Tuple[str, str]],
+    typing.Mapping[str, typing.Optional[str]],
+]
+CookiesType = typing.Union[
+    typing.Mapping[str, str], Cookies,
+]
+DataType = typing.Union[
+    typing.AnyStr,
+    typing.BinaryIO,
+    typing.TextIO,
+    typing.Iterable[typing.AnyStr],
+    # This needs to get unasync-ed into 'typing.Iterable[typing.AnyStr]'
+    typing.AsyncIterable[typing.AnyStr],
+    # This needs to get unasync-ed into 'SyncRequestData'
+    AsyncRequestData,
+]
+JSONType = typing.Union[
+    typing.Mapping["JSONType", "JSONType"],
+    typing.Sequence["JSONType"],
+    int,
+    bool,
+    str,
+    float,
+    None,
+]
+
+
+class Session:
+    """
+    The central instance that manages HTTP life-cycles and interfaces
+    with the background connection pools.
+
+    Adding all of the shortened 'per-method' functions to the
+    Session can be done later once the entire interface is complete.
+    Until that time they are basically just dead-weight for testing
+    and updating.
+    """
+
+    def __init__(self, trust_env: bool = True):
+        self.trust_env = trust_env
+
+    async def request(
+        self,
+        # Request Headers
+        method: str,
+        url: str,
+        headers: HeadersType = None,
+        auth=None,
+        cookies: CookiesType = None,
+        query: QueryType = None,
+        # Request Body
+        data: DataType = None,
+        json: JSONType = None,
+        # Connection
+        timeout=None,
+        proxies=None,
+        http_versions: typing.Sequence[str] = None,
```

Comment:
Maybe add a comment that we only support HTTP/1.1 currently? `HTTP/1.1` is easy to grep for and will help when we'll have to look at the places to update to add HTTP/2 support.